src/libraries/System.Linq.Expressions/src/ILLink/ILLink.Substitutions.xml

```diff
@@ -0,0 +1,13 @@
+<linker>
+  <assembly fullname="System.Linq.Expressions">
+    <type fullname="System.Linq.Expressions.LambdaExpression">
+      <method signature="System.Boolean get_CanCompileToIL()" feature="System.Linq.Expressions.CanCompileToIL" featurevalue="false" body="stub" value="false" />
+    </type>
+    <type fullname="System.Dynamic.Utils.DelegateHelpers">
+      <method signature="System.Boolean get_CanEmitObjectArrayDelegate()" feature="System.Linq.Expressions.CanEmitObjectArrayDelegate" featurevalue="false" body="stub" value="false" />
+    </type>
+    <type fullname="System.Linq.Expressions.Interpreter.CallInstruction">
+      <method signature="System.Boolean get_CanCreateArbitraryDelegates()" feature="System.Linq.Expressions.CanCreateArbitraryDelegates" featurevalue="false" body="stub" value="false" />
+    </type>
```

Comment:
Property living in Expressions was the last direction https://github.com/dotnet/runtime/issues/17973#issuecomment-277144969 discussed. I'm just following that, given the API is unreviewed.

get_CanEmitObjectArrayDelegate and get_CanCreateArbitraryDelegates probably won't be APIs ever. CanCompileToIL might.

Reply:
I don't understand exactly what they try to do but it seems to me they react to runtime limitations, right? Should not this be then call to runtime to get the actual value with a feature switch on top of it?

Reply:
`get_CanEmitObjectArrayDelegate` and `get_CanCreateArbitraryDelegates` could probably be replaced with `RuntimeFeature.IsDynamicCodeSupported` _eventually_, but it would be a behavior change and this pull request didn't intend to do behavior changes, just replace `#define`s with something that can be overriden.

Replacing `get_CanCreateArbitraryDelegates` with `IsDynamicCodeSupported` would make the iDevices version take the codepaths that correspond to `FEATURE_DLG_INVOKE` not being defined in the csproj and that's not how iDevices build was set up in #54970. If you would like me to make that change, I can do it, but I cannot investigate any possible fallouts from it because I don't work on or own iDevices. An observable difference will be slower perf when invoking method call instructions in the interpreter. An upside will be that some of the tests that were blocked in #54970 will work. Ideally all of this (configuring the library so that it properly works, and revisiting tests blocked in #54970) should happen in subsequent pull requests. I definitely plan to revisit the blocked tests.

`get_CanEmitObjectArrayDelegate` cannot be replaced with `RuntimeFeature.IsDynamicCodeSupported` at this moment because it requires a runtime API that is not implemented on Mono and would therefore break iDevices (it's an API to create a strongly typed Delegate that boxes all it's arguments, puts them into an array, dispatches to a `Func<object?[], object?>` delegate to do the actual work, and unboxes the result - doing all of that without runtime codegen).